input/vlct/dual_energy_cloud/initial_cloud.in

```diff
@@ -0,0 +1,264 @@
+	 value = 8.944866290871940e-02 * 0.01295;
+     };
+
+     velocity_x_upwind {
+         type = "inflow";
+         axis = "x";
+         face = "lower";
+         field_list = "velocity_x";
+         value = 1.341500614584355e+01;
+     };
+
+     transverse_velocity_upwind {
+         type = "inflow";
+         axis = "x";
+	  face = "lower";
+         field_list = ["velocity_y","velocity_z"];
+         value = 0.;
+     };
+
+     etot_upwind{
+         type = "inflow";
+         axis = "x";
+         face = "lower";
+         field_list = "total_energy";
+         value = 1.619661509037183e+02;
+     };
+
+     eint_upwind{
+         type = "inflow";
+         axis = "x";
+         face = "lower";
+         field_list = "internal_energy";
+         value = 7.198495595720811e+01;
+     };
+
+     bfield_upwind{
+         type = "inflow";
+         axis = "x";
+         face = "lower";
+         field_list = ["bfield_x", "bfieldi_x",
+                       "bfield_y", "bfieldi_y",
+                       "bfield_z", "bfieldi_z"];
+         value = 0.;
+     }
+
+     cloud_dye_upwind{
+         type = "inflow";
+         axis = "x";
+         face = "lower";
+         field_list = "cloud_dye";
+         value = 0.0;
+     };
+
+     yedge {
+         type = "outflow";
+     };
+
+     zedge {
+         type = "outflow";
+     };
+ }
+
+ Stopping {
+     time =  0.0625;
+ }
+
+Units {
+     #density = 10^-3 mH/cm^3
+     density = 1.6737352238051867e-27; # g / cm^3
+     length   = 4.629e+22; # the cloud radius
+     time     = 1.055968875226116e+16; # the cloud-crushing time
+}
+
+#======================================================================
+# DISCRETIZATION
+#======================================================================
+
+ Mesh {
+     root_rank = 3;
+     # 8 cells per cloud radius
+     root_size   = [ 32, 32, 32];
+     root_blocks = [  1,  1,  1];  # number of blocks in root grid
+ }
+
+ Field {
+     # The VL+CT method tracks primitive quantities.
+     #    - bfieldi_x/y/z are interface magnetic fields (aka longitudinal).
+     #      The x/y/z component is the only component tracked at the faces
+     #      between cells along the x/y/z component
+     #    - bfield_x/y/z are the cell-centered magenetic fields. Each
+     #      component is the average value of the corresponding interface
+     #      values
+     list = ["density", "velocity_x", "velocity_y", "velocity_z",
+             "cloud_dye", "metal_density", "temperature", "pressure", 
+	     "total_energy", "internal_energy",
+             "bfieldi_x", "bfieldi_y", "bfieldi_z",
+             "bfield_x", "bfield_y", "bfield_z"];
+
+     gamma = 1.6666666666666667;
+
+     # The VL+CT method reconstructs interface values (to compute fluxes) twice
+     # per timestep. Can perform nearest-neighbor or PLM. Each time
+     # nearest-neighbor is used, 1 ghost zone is needed. Each time PLM is used
+     # 2 ghost zones are needed.
```

Comment:
Is this statement true for the implementation here? I also saw this in the table in the docs but though I may have misunderstood.
NN+PLM should be fine with using 2 ghost cells total, doesn't it?
Both NN and PLM only use an `i-1` and/or `i+1` cell for the reconstruction at `i` so for a two step integrator 2 ghost cells per cycle should cover it.

Reply:
It's not a typo. I find this topic to be intuitive, and easiest to think about it terms of examples. Consider the following 5-element array:

```
+-----+-----+-----+-----+-----+
| i=0 | i=1 | i=2 | i=3 | i=4 |
|     |     |     |     |     |
+-----+-----+-----+-----+-----+
```

For NN, we can reconstruct the values on both the left and right sides if i = 0.5, 1.5,2.5,3.5. This obviously allows us to compute the fluxes at i = 0.5, 1.5,2.5,3.5.

For PLM, we can reconstruct the values:
- on the left sides of i = 1.5, 2.5, 3.5
- on the right sides of i = 0.5, 1.5, 2.5

Therefore we can only compute the fluxes at i = 1.5 and i = 2.5 using PLM. I think this illustrates that PLM requires 1 more ghost zone than NN

Reply:
My bad. I got confused between this VL+CT implementation and another MUSCL Hancock integrator, which gets second order accuracy using PLM and only 2 ghost cells per step.